eoscompanion/routes.py

```diff
@@ -161,19 +160,419 @@ def desktop_id_to_app_id(desktop_id):
+    EosDiscoveryFeed.CardStoreType.WORD_QUOTE_CARD: _serialize_word_quote_content_feed_model
+}
+
+def content_feed_model_to_json_entries(model, app_id, query):
+    '''Return a list of json entries for each model.
+
+    This varies depending on the model type. Some models may return
+    more than one entry (for instance, the word-quote model returns
+    two entries).
+    '''
+    return _CONTENT_FEED_MODEL_SERIALIZERS[model.get_property('type')](model,
+                                                                       app_id,
+                                                                       query)
+
+
+def entries_from_content_feed_models(models, sources, query):
+    '''Generate the "entries" entry for the feed JSON response.
+
+    This response will contain every model in the models as long as
+    the model type is known and its source was present in the sources
+    entry.
+
+    Problems with individual models will be reported on the console.
+    '''
+    # A map of sources, first by type and then by a unique identifier
+    # for the source of that type. This is used to quickly lookup whether
+    # an entry has a corresponding source.
+    sources_map = defaultdict(set)
+
+    for source in sources:
+        sources_map[source['type']].add(get_source_identifier(source))
+
+    for model in models:
+        # Get the app ID for all content models that have a desktop ID. If
+        # a content model does not have a desktop ID then we can still
+        # display it, but the user won't be able to navigate to content
+        # for that app.
+        try:
+            app_id = desktop_id_to_app_id(model.get_property('desktop-id'))
+        except TypeError:
+            app_id = None
+
+        if app_id is not None and app_id not in sources_map['application']:
+            logging.warning('Model %s with app-id %s did not have a '
+                            'corresponding entry in the sources list, it will be '
+                            'ignored', model, app_id)
+            continue
+
+        for entry in content_feed_model_to_json_entries(model, app_id, query):
+            yield entry
+
+
 @require_query_string_param('deviceUUID')
 @require_query_string_param('mode')
 @record_metric('af3e89b2-8293-4703-809c-8e0231c128cb')
-def companion_app_server_feed_route(server, msg, path, query, *args):
-    '''Send feed.'''
-    del server
+def companion_app_server_feed_route(server,
+                                    msg,
+                                    path,
+                                    query,
+                                    context,
+                                    content_db_conn):
+    '''Request the Content Feed from EosDiscoveryFeed.
+
+    The :mode: paramter is used to determine whether newer entries
+    should be fetched or older entries should be fetched. It is currently
+    unused for now, the feed is the same as the Discovery Feed on the
```

Comment:
Either remove `currently` or remove `for now` (I'd remove `currently` since you use `for now` below as well)

Reply:
Actually, repeating adjectives is a little mental bug I seem to have. Thanks for catching, I removed the last "for now".